### PR TITLE
Document the WebDAV Custom Groups Management API

### LINKED ISSUE #413: Document Custom Groups WebDAV API
Custom Groups already have a WebDAV API which until now was private and undocumented. As there's demand for it, we want to declare it public and add it to the official documentation. 

Input: https://github.com/owncloud/customgroups/issues/3

@settermjd @PVince81

### PR DESCRIPTION
**Pull request by settermjd**
This relates to https://github.com/owncloud/customgroups/issues/3 and fixes #413.

### CODE REVIEW

File: modules/developer_manual/examples/core/webdav_api/group/report-customgroups.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<oc:filter-members xmlns:a="DAV:" xmlns:oc="http://owncloud.org/ns" >
+	<oc:filter-rules>
+		<oc:user-id>admin</oc:user-id>
+		<!--	<oc:is-admin>1</oc:is-admin> -->
```
Comment (PVince81) [inline modules/developer_manual/examples/core/webdav_api/group/report-customgroups.xml]: is this comment intentional ?

File: modules/developer_manual/pages/_partials/webdav_api/responses/insufficient-privileges-overview.adoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
+====== Insufficient Privileges
+
+If the user making the request has insufficient privileges to make the request then a status of `HTTP/1.1 401 Unauthorized` will be returned, along with the following XML in the response’s body:
```
Comment (PVince81) [inline modules/developer_manual/pages/_partials/webdav_api/responses/insufficient-privileges-overview.adoc]: In general it should be 403, for example if you try to perform a request as regular user that only admins can do.

The 401 is whenever the user cannot login / auth.

File: modules/developer_manual/examples/core/webdav_api/group/report-customgroups.xml (hunk 1 above)
Comment (settermjd) [reply modules/developer_manual/examples/core/webdav_api/group/report-customgroups.xml]: No, I don't believe so.

File: modules/developer_manual/pages/_partials/webdav_api/responses/insufficient-privileges-overview.adoc (hunk 2 above)
Comment (settermjd) [reply modules/developer_manual/pages/_partials/webdav_api/responses/insufficient-privileges-overview.adoc]: @PVince81 can you review again please?

File: modules/developer_manual/examples/core/webdav_api/group/request/list-custom-groups.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<oc:customgroups 
+    xmlns:a="DAV:" 
+    xmlns:oc="http://owncloud.org/ns">
+    <a:prop>
+        <a:getlastmodified />
+        <a:getlastmodified/>
```
Comment (PVince81) [inline modules/developer_manual/examples/core/webdav_api/group/request/list-custom-groups.xml]: duplicate property

File: modules/developer_manual/pages/webdav_api/groups/custom_groups_endpoints.adoc [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+....
+
+=== Rename Custom Group
+:request_method: PROPPATCH
+:request_data_file: rename-custom-group.xml
+:request_path_suffix: <$groupId>
+
+This endpoint allows a custom group to be renamed.
+
+NOTE: Only group admins can rename a group.
```
Comment (PVince81) [inline modules/developer_manual/pages/webdav_api/groups/custom_groups_endpoints.adoc]: they can rename the groups they manage, not sure if it's clear in this sentence

Comment (settermjd) [reply modules/developer_manual/pages/webdav_api/groups/custom_groups_endpoints.adoc]: Just updated that sentence. I hope it's sufficiently clear now.

**Pull request merged**